src/v/utils/utf8.h

```diff
@@ -11,8 +11,14 @@
 
 #pragma once
 
+// clang-format off
+// encoding_utf does not include std::basic_string and will fail compilation
+// if <string> isn't included before it
+#include <string>
+// clang-format on
 #include <boost/locale/encoding_utf.hpp>
```

Comment:
I think the way to solve this is to replace the existing utf header with `#include <boost/locale.hpp>`

Reply:
Actually putting `#include "likely.h" above it seemed to have solved it...